FlowCrypt/Controllers/Attachment/AttachmentViewController.swift

```diff
@@ -56,22 +54,27 @@ final class AttachmentViewController: UIViewController {
         super.viewDidLoad()
         setupNavigationBar()
         addWebView()
-
-        filesManager.save(file: file)
-            .sink(
-                receiveCompletion: {_ in},
-                receiveValue: { [weak self] url in
-                    self?.load(with: url)
-                }
-            )
-            .store(in: &cancellable)
+        showSpinner()
+        title = file.name
+        saveAndStartDownload(file: file)
     }
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
-        filesManager.remove(file: file)
-            .sink(receiveCompletion: {_ in}, receiveValue: {_ in})
-            .store(in: &cancellable)
+        remove(file: file)
+    }
+
+    private func remove(file: FileType) {
+        Task {
+            try await filesManager.remove(file: file)
+        }
+    }
```

Comment:
There should be a try/catch and a handle with modal showing the error (don't ignore errors)